pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -231,31 +269,61 @@ func TestDesiredRouterDeployment(t *testing.T) {
 		t.Errorf("router Deployment has unexpected toleration: %#v",
 			deployment.Spec.Template.Spec.Tolerations)
 	}
+	tests := []envData{
+		{WildcardRouteAdmissionPolicy, true, "false"},
+		{"NAMESPACE_LABELS", true, "foo=bar"},
+		{"ROUTE_LABELS", true, "baz=quux"},
+	}
+	if err := checkDeploymentEnvironment(t, deployment, tests); err != nil {
+		for _, e := range err {
+			t.Errorf("%s", e)
+		}
+	}
+}
 
-	checkDeploymentHasEnvVar(t, deployment, "ROUTER_HAPROXY_CONFIG_MANAGER", false, "")
-
-	checkDeploymentHasEnvVar(t, deployment, "ROUTER_LOAD_BALANCE_ALGORITHM", true, "leastconn")
-	checkDeploymentHasEnvVar(t, deployment, "ROUTER_TCP_BALANCE_SCHEME", true, "source")
-	checkDeploymentDoesNotHaveEnvVar(t, deployment, "ROUTER_ERRORFILE_503")
-	checkDeploymentDoesNotHaveEnvVar(t, deployment, "ROUTER_ERRORFILE_404")
-
-	checkDeploymentHasEnvVar(t, deployment, "ROUTER_MAX_CONNECTIONS", false, "")
+func TestDesiredRouterDeploymentDefaultEnv(t *testing.T) {
```

Comment:
It appears that `TestDesiredRouterDeploymentDefaults`, `TestDesiredRouterDeploymentDefaultEnv`, and `TestDesiredRouterDeploymentSpecTemplate` are verifying various subsets of the parameters of the same deployment.  Is it useful to split these into 3 tests instead of having 1 test for each deployment?

Reply:
I'm missing something. Do you mean you want to combine these into one test?  They are already split into 3 tests.

Reply:
Yeah, I was wondering why they are in 3 separate tests when each test is creating the same router deployment (same inputs, same output) and then testing some subset of environment variables.

Reply:
Ok, I combined all the env var checking in `TestDesiredRouterDeployment` and removed `TestDesiredRouterDeploymentDefaultEnv`